### ISSUE TITLE
iPad Mini 4 WiFi (Model A1538) iOS 14+ jailbreak issue

### ISSUE THREAD

**Issue by dzrdmwjx** [checkra1n/BugTracker#1940]
**Tell us about your setup:**
1. What iDevice are you using? iPad Mini4 WiFi (Model A1538)
2. On what version of iOS is it? iOS 14.2, 14.3 (tried both)
3. What version of checkra1n are you using? 0.12.1, 0.12.2 (tried both)
4. What is your host system (OS version? Hackintosh? VM? etc.)? Linux
5. How are you connecting to the device (USB-A? USB-C? Apple/3rd party cable? Through a USB hub?)? USB-A

**What are the steps to reproduce the issue?**
1. Connect the device to the host and let the checkra1n app detect it
2. Proceed with jailbreak and let checkra1n put the device into recovery mode
3. Follow the app instructions to manually put the device into DFU mode and let the app continue with the jailbreak process
...

**What do you expect, and what is happening instead?**
The device should reboot, display the checkra1n / PongoOS splash screen and boot up into the jailbroken mode.
What happens instead is that the PongoOS splash screen doesn't show up at all and the device boots up in the normal mode with checkra1n displaying USBMUXD Error -79

**Did you see a popup on the device stating it entered "Safe Mode" due to an error?**
No

**Does the issue also occur if you tick "Safe Mode" in the checkra1n options?**
Yes

**Any other info, error logs, screenshots, ...?**

It's worth noting that the above process randomly works if I just keep repeating these steps. Mostly it takes roughly 10-15 tries, but on rare occasions, I have had success with as few as 1-2 tries while at other times, it required 20+ tries. So it's not that checkra1n doesn't jailbreak this device at all, but it's just that it's too unreliable to be practically useful (it's not reasonable to put the device in DFU up to 10-15 times for a successful jailbreak every time it powers off).

I enabled verbose logging and compared the pass and fail logs and there's no difference in the logs well beyond the the point of failure (that is, when the checkra1n icon and PongoOS splash normally show up). It's only after the device is completely booted up that USBMUX error is displayed in checkra1n. Note that even when the jailbreak is successful, checkra1n still displays USBMUX error, but with a different code (-77). If there's a way to capture additional diagnostic information, I will be happy to assist but verbose logging is all I can think of right now, which doesn't tell much as to why checkm8 exploit is failing.

I have tried all the suggestions to get a more reliable result - including different ports on the host computer, official cable, safe mode setting, but nothing seems to make a difference. 

Success log:
```
 - [01/02/21 12:52:52] <Info>: Booting...
 - [01/02/21 12:52:52] <Verbose>: Setting bootargs to: rootdev=md0
 - [01/02/21 12:53:17] <Verbose>: Found USBMux interface /sys/devices/pci0000:00/0000:00:14.0/usb2/2-1/2-1:4.1 with endpoints 85 and 04
 - [01/02/21 12:53:17] <Error>: Failed to claim USB device interface: (Resource busy) (error code: -77)

```
Failure log:
```
 - [01/02/21 12:42:14] <Info>: Booting...
 - [01/02/21 12:42:14] <Verbose>: Setting bootargs to: rootdev=md0
 - [01/02/21 12:43:10] <Verbose>: Using usbmuxd
 - [01/02/21 12:43:13] <Warning>: connection refused?!
 - [01/02/21 12:43:13] <Error>: usbmux: Failed to connect (error code: -79)

```

**Comment by dzrdmwjx** [comment]
For what it's worth, a number of people have reported a similar issue with their iPad Air 2 devices on iOS 14+, where they follow the process to the dot but the there's no checkra1n / PongoOS splash screen and the device boots up normally with checkra1n displaying USBMUX -79 in case of linux or USBMUX -20 in case of macos. I'm not sure if that is the same issue as what I'm experiencing with my iPad Mini 4, but it's worth noting that both are A8/A8X devices.

I should also note that with the same setup, I do not have any problem jailbreaking my iPhone 6 running on iOS 12 and even this device until I upgraded it to IOS 14.

**Comment by pasha5230** [comment]
you only need to write to developers.no one reads here

**Comment by ghost** [comment]
Yes this needs to be addressed

**Comment by dzrdmwjx** [comment]
I no longer have this issue with the recently released checkra1n 0.12.3 with updated PongoOS even after upgrading to iOS 14.5.

_Originally posted by @dzrdmwjx in https://github.com/checkra1n/BugTracker/issues/1880#issuecomment-830641537_
